### PR TITLE
[ workflow ] Simplify stack.yml using stack-root output

### PR DESCRIPTION
**Pull request by andreasabel**
Try: simplify `stack.yml` using `stack-root` output from `haskell-setup` action.

Squash before merging.

### CODE REVIEW

File: .github/workflows/stack.yml [hunk 1]
```diff
@@ -60,31 +60,38 @@ jobs:
...
+    - run: |
+        echo "STACK_ROOT (fix)  = ${STACK_ROOT}"
+        echo "STACK_VER         = ${STACK_VER}"
+      name: More information about the Haskell setup
     - run: |
         stack update --silent
-        if [[ "$OSTYPE" == "linux-gnu"* || "$OSTYPE" == "darwin"* ]]; then
-          echo "system-ghc: true" >> ~/.stack/config.yaml
-        else
-          echo "system-ghc: true" >> ${STACK_ROOT}/config.yaml
```
Comment (L-TChen) [inline .github/workflows/stack.yml:68]: Oops, I thought the environment variable `STACK_ROOT` was windows-specific. Good Job.

**Review by L-TChen — APPROVED** [review]
It looks good to me!

File: .github/workflows/stack.yml (hunk 1 above)
Comment (andreasabel) [reply .github/workflows/stack.yml:68]: Well, you were right!  Indeed `STACK_ROOT` is set only on Windows.  This is why I have added some steps to set it on all runners, using the `stack-root` output from the setup action.

Comment (L-TChen) [reply .github/workflows/stack.yml:68]: A-ha. I see. The previous step does not just print information but also set up this variable.

**Pull request merged**